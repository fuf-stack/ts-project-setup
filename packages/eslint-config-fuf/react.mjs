import { configs, rules } from 'eslint-config-airbnb-extended';
import jsxA11yPlugin from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooksPlugin from 'eslint-plugin-react-hooks';

import baseConfig from './base.mjs';
import fufStorybook from './fuf/fuf-react-storybook.mjs';
import fufReact from './fuf/fuf-react.mjs';

export default [
  // Base Config
  ...baseConfig,

  // Register React-related plugins explicitly for flat config resolution
  {
    name: 'fuf/react/plugins',
    plugins: {
      react: reactPlugin,
      'react-hooks': reactHooksPlugin,
      'jsx-a11y': jsxA11yPlugin,
    },
  },

  // Airbnb React Recommended Config
  ...configs.react.recommended,
  // Airbnb React TypeScript Config
  ...configs.react.typescript,
  // Strict React Config
  rules.react.strict,

  // FUF React Configs
  fufReact,
  fufStorybook,
];
