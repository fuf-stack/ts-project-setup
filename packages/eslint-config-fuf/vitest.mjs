// see: https://github.com/veritem/eslint-plugin-vitest

import vitest from '@vitest/eslint-plugin';

export default [
  {
    name: 'fuf/vitest',
    plugins: { vitest },

    // configure file patterns for tests
    files: ['**/*.spec.ts', '**/*.spec.tsx', '**/*.test.ts', '**/*.test.tsx'],

    // test file rules
    rules: {
      // recommended vitest rules
      ...vitest.configs.recommended.rules,

      // Turn off legacy import plugin rules if present upstream
      'import-x/extensions': 'off',
      'import/extensions': 'off',

      // Do not enforce production dependencies in tests
      'import/no-extraneous-dependencies': 'off',
      'import-x/no-extraneous-dependencies': 'off',
    },
  },
];
