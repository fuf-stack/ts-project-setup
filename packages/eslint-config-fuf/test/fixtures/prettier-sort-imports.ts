/* eslint-disable */

/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import-x/extensions */
/* eslint-disable import-x/no-unresolved */
/* eslint-disable no-unused-vars */

import moduleStyles from './Some.module.css';
import jsonFile from './data.json';
import fs from 'node:fs';
import react from 'react';

import relative from '../relative/path';
import png from './image.png';
import alsoAbsolute from 'src/also/absolute';
import someOtherThing from '@fuf-stack/some-other-thing';
import { describe, expect, it } from 'vitest';
import jpg from './image.jpg';
import path from 'node:path';

import svg from './image.svg';
import thirdParty from 'third-party-module';
import something from '@fuf-stack/something';
import absolute from 'src/absolute';
import type { ReactElement } from 'react';
import jpeg from './image.jpeg';

// INFO: css imports will not be touched
import 'some-module/style.css';

const MyComponent = (): ReactElement => null;

export default MyComponent;
