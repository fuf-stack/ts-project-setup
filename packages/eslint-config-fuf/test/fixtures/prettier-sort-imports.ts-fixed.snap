import type { ReactElement } from 'react';

import { describe, expect, it } from 'vitest';

import fs from 'node:fs';
import path from 'node:path';

import react from 'react';

import thirdParty from 'third-party-module';

import someOtherThing from '@fuf-stack/some-other-thing';
import something from '@fuf-stack/something';

import absolute from 'src/absolute';
import alsoAbsolute from 'src/also/absolute';

import relative from '../relative/path';

import jsonFile from './data.json';

import jpeg from './image.jpeg';
import jpg from './image.jpg';
import png from './image.png';
import svg from './image.svg';

import moduleStyles from './Some.module.css';

// INFO: css imports will not be touched
import 'some-module/style.css';

const MyComponent = (): ReactElement => null;

export default MyComponent;
